src/contracts/core/AVSDirectory.sol

```diff
@@ -581,42 +586,40 @@ contract AVSDirectory is
             // 3. ensure only queued deallocation per operator, operatorSet, strategy
             _checkPendingDeallocations(operator, strategy, operatorSets[i]);
 
-            // 4. queue deallocation for (op, opSet, strategy) for magnitudeDiff amount
-            _queuedDeallocations[operator][strategy][operatorSets[i].avs][operatorSets[i].operatorSetId].push(
-                QueuedDeallocation({
+            // 4. push PendingFreeMagnitude and respective array index into (op,opSet,Strategy) queued deallocations
+            uint256 index = _pendingFreeMagnitude[operator][strategy].length;
+            _pendingFreeMagnitude[operator][strategy].push(
+                PendingFreeMagnitude({
                     magnitudeDiff: deallocation.magnitudeDiffs[i],
                     completableTimestamp: completableTimestamp
                 })
             );
+            _queuedDeallocationIndices[operator][strategy][operatorSets[i].avs][operatorSets[i].operatorSetId].push(index);
```

Comment:
imo its time combine `_magnitudeUpdates` and `_queuedDeallocationIndices`

Reply:
opting to keep separate mappings otherwise lookup for indices in slashOperator would require looking through future magnitudeUpdates as well